unit/analyses/ai/ai_simplify_lhs.cpp

```diff
@@ -33,6 +33,10 @@ class constant_simplification_mockt:public ai_domain_baset
   {}
   void make_entry() override
   {}
+  bool is_bottom() const override
+  { return true; }
```

Comment:
Probably this should be `return false` so the mock is at least consistent? Alternatively, an `UNREACHABLE` so that the test isn't using it.

Reply:
It is not, formally speaking, wrong for an abstraction to be both top and bottom at the same time.  It means that it is an abstraction that only has one possible state.

But, yeah, UNREACHABLE as well would be good.

Reply:
Fixed.